The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4968736986/jobs/8891514545) [(plain)](https://github.com/rust-lang/rust/commit/5fd4b65c1591c4ae8961d88e5da0f52bed4bc263/checks/13458308155/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling object v0.30.1
   Compiling hashbrown v0.13.1
   Compiling miniz_oxide v0.6.2
   Compiling addr2line v0.19.0
error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1174 |         self.bytes().eq(other.bytes())
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1174 |         self.bytes().eq(other.bytes())
     |                               ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1201 |         self.bytes().partial_cmp(other.bytes())
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1201 |         self.bytes().partial_cmp(other.bytes())
     |                                        ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1205 |         self.bytes().lt(other.bytes())
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1205 |         self.bytes().lt(other.bytes())
     |                               ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1209 |         self.bytes().le(other.bytes())
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1209 |         self.bytes().le(other.bytes())
     |                               ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1213 |         self.bytes().gt(other.bytes())
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1213 |         self.bytes().gt(other.bytes())
     |                               ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1217 |         self.bytes().ge(other.bytes())
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1217 |         self.bytes().ge(other.bytes())
     |                               ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1236 |         self.bytes().cmp(other.bytes())
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1236 |         self.bytes().cmp(other.bytes())
     |                                ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1286 |         self.bytes().hash(state)
     |              ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
    |
196 |             s.bytes().len()
    |               ^^^^^ method not found in `&OsStr`
    |
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
    |
    |
333 |     if file.bytes() == b".." {
    |             ^^^^^ method not found in `&OsStr`
    = help: items from traits can only be used if the trait is implemented and in scope
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
    |
    |
341 |     let mut iter = file.bytes().rsplitn(2, |b| *b == b'.');
    |                         ^^^^^ method not found in `&OsStr`
    = help: items from traits can only be used if the trait is implemented and in scope
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
    |
352 |     let slice = file.bytes();
    |                      ^^^^^ method not found in `&OsStr`
    |
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1481 |             Some(f) => f.bytes(),
     |                          ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for struct `OsString` in the current scope
     |
     |
1486 |         let start = self.inner.bytes().as_ptr().addr();
     |                                ^^^^^ method not found in `OsString`
    ::: library/std/src/ffi/os_str.rs:91:1
     |
91   | pub struct OsString {
91   | pub struct OsString {
     | ------------------- method `bytes` not found for this struct
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
     |
     |
1491 |         let new = extension.bytes();
     |                             ^^^^^ method not found in `&OsStr`
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for struct `OsStr` in the current scope
     |
2015 |         self.inner.bytes()
     |                    ^^^^^ method not found in `OsStr`
     |
     |
    ::: library/std/src/ffi/os_str.rs:119:1
     |
119  | pub struct OsStr {
     | ---------------- method `bytes` not found for this struct
     = help: items from traits can only be used if the trait is implemented and in scope
     = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
     |
568  | pub trait Read {
     | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
   --> library/std/src/sys/common/small_c_string.rs:22:36
    |
22  |     run_with_cstr(path.as_os_str().bytes(), f)
    |                                    ^^^^^ method not found in `&OsStr`
    = help: items from traits can only be used if the trait is implemented and in scope
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
    |
    |
33  |     let path_os = path.as_os_str().bytes();
    |                                    ^^^^^ method not found in `&OsStr`
    = help: items from traits can only be used if the trait is implemented and in scope
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
    |
    |
167 |         if program.bytes().starts_with(b"/") {
    |                    ^^^^^ method not found in `&OsStr`
    = help: items from traits can only be used if the trait is implemented and in scope
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^


error[E0599]: no method named `bytes` found for reference `&OsStr` in the current scope
    |
    |
169 |         } else if program.bytes().contains(&b'/') {
    |                           ^^^^^ method not found in `&OsStr`
    = help: items from traits can only be used if the trait is implemented and in scope
    = help: items from traits can only be used if the trait is implemented and in scope
note: `io::Read` defines an item `bytes`, perhaps you need to implement it
    |
568 | pub trait Read {
    | ^^^^^^^^^^^^^^

```

</details>

--- a/library/std/src/ffi/os_str.rs
+++ b/library/std/src/ffi/os_str.rs
@@ -892,7 +892,7 @@ impl OsStr {
     /// Note: As the encoding is unspecified, any sub-slice of bytes that is not valid UTF-8 should
     /// be treated as opaque and only comparable within the same rust version built for the same
     /// target platform.  For example, sending the slice over the network or storing it in a file
-    /// will likely result in incompatible byte slices  See [`OsString`] for more encoding details
+    /// will likely result in incompatible byte slices.  See [`OsString`] for more encoding details
     /// and [`std::ffi`] for platform-specific, specified conversions.
     ///
     /// [`std::ffi`]: crate::ffi

--- a/library/std/src/ffi/os_str.rs
+++ b/library/std/src/ffi/os_str.rs
@@ -670,18 +670,22 @@ impl OsStr {
     /// Converts a slice of bytes to an OS string slice without checking that the string contains
     /// valid `OsStr`-encoded data.
     ///
+    /// The byte encoding is an unspecified, platform-specific, self-synchronizing superset of UTF-8.
+    /// By being a self-synchronizing superset of UTF-8, this encoding is also a superset of 7-bit
+    /// ASCII.
+    ///
     /// See the [module's toplevel documentation about conversions][conversions] for safe,
     /// cross-platform [conversions] from/to native representations.
     ///
     /// # Safety
     ///
-    /// `OsStr`'s encoding is an unspecified superset of UTF-8 and callers must
-    /// pass in bytes that originated as a mixture of validated UTF-8 and bytes from
-    /// [`OsStr::as_os_str_bytes`] from within the same rust version built for the same target
-    /// platform.  For example, reconstructing an `OsStr` from bytes sent over the network or stored
-    /// in a file will likely violate these safety rules.  The bytes from `OsStr::as_os_str_bytes`
-    /// may be split either immediately before or immediately after some valid non-empty UTF-8
-    /// substring
+    /// As the encoding is unspecified, callers must pass in bytes that originated as a mixture of
+    /// validated UTF-8 and bytes from [`OsStr::as_os_str_bytes`] from within the same rust version
+    /// built for the same target platform.  For example, reconstructing an `OsStr` from bytes sent
+    /// over the network or stored in a file will likely violate these safety rules.
+    ///
+    /// Due to the encoding being self-synchronizing, the bytes from [`OsStr::as_os_str_bytes`] can be
+    /// split either immediately before or immediately after any valid non-empty UTF-8 substring.
     ///
     /// # Example
     ///
@@ -880,6 +884,10 @@ impl OsStr {
 
     /// Converts an OS string slice to a byte slice.  To convert the byte slice back into an OS
     /// string slice, use the [`OsStr::from_os_str_bytes_unchecked`] function.
+    ///
+    /// The byte encoding is an unspecified, platform-specific, self-synchronizing superset of UTF-8.
+    /// By being a self-synchronizing superset of UTF-8, this encoding is also a superset of 7-bit
+    /// ASCII.
     ///
     /// Note: As the encoding is unspecified, any sub-slice of bytes that is not valid UTF-8 should
     /// be treated as opaque and only comparable within the same rust version built for the same

--- a/library/std/src/ffi/os_str.rs
+++ b/library/std/src/ffi/os_str.rs
@@ -678,8 +678,10 @@ impl OsStr {
     /// `OsStr`'s encoding is an unspecified superset of UTF-8 and callers must
     /// pass in bytes that originated as a mixture of validated UTF-8 and bytes from
     /// [`OsStr::as_os_str_bytes`] from within the same rust version built for the same target
-    /// platform.  The bytes from `OsStr::as_os_str_bytes` may be split either
-    /// immediately before or immediately after some valid non-empty UTF-8 substring
+    /// platform.  For example, reconstructing an `OsStr` from bytes sent over the network or stored
+    /// in a file will likely violate these safety rules.  The bytes from `OsStr::as_os_str_bytes`
+    /// may be split either immediately before or immediately after some valid non-empty UTF-8
+    /// substring
     ///
     /// # Example
     ///
@@ -881,8 +883,9 @@ impl OsStr {
     ///
     /// Note: As the encoding is unspecified, any sub-slice of bytes that is not valid UTF-8 should
     /// be treated as opaque and only comparable within the same rust version built for the same
-    /// target platform.  See [`OsString`] for more encoding details and [`std::ffi`] for
-    /// platform-specific, specified conversions.
+    /// target platform.  For example, sending the slice over the network or storing it in a file
+    /// will likely result in incompatible byte slices  See [`OsString`] for more encoding details
+    /// and [`std::ffi`] for platform-specific, specified conversions.
     ///
     /// [`std::ffi`]: crate::ffi
     #[inline]

